cfgov/login/forms.py

```diff
@@ -53,101 +52,104 @@ class CFGOVSetPasswordForm(PasswordValidationMixin, SetPasswordForm):
 
 
 class LoginForm(AuthenticationForm):
+    error_messages = {
+        "temporary_lock": (
+            "This account is temporarily locked; please try later or "
+            '<a href="/admin/password_reset/" style="color:white;font-weight:bold">'
+            "reset your password</a>."
+        ),
+        "password_expired": (
+            "Your password has expired. Please "
+            '<a href="/admin/password_reset/" style="color:white;font-weight:bold">'
+            "reset your password</a>."
+        ),
+        **AuthenticationForm.error_messages,
+    }
+
     def clean(self):
-        username = self.cleaned_data.get("username")
-        password = self.cleaned_data.get("password")
+        try:
+            return super().clean()
+        except ValidationError as e:
+            # The Django AuthenticationForm raised a validation error because
+            # it could not authenticate the username/password combination.
+            #
+            # We want to record the failed login attempt, and maybe raise a
```

Comment:
This change alters the login behavior when the user is locked out.

[In the existing code](https://github.com/cfpb/consumerfinance.gov/blob/cbd6b027516fc1644303f38a42fbeae82e7777bb/cfgov/login/forms.py#L77-L79), if a user successfully authenticates, but is locked out based on previous login failures, then the login [will be blocked](https://github.com/cfpb/consumerfinance.gov/blob/cbd6b027516fc1644303f38a42fbeae82e7777bb/cfgov/login/forms.py#L143-L149) with a "temporarily locked" message. A new login failure [will not](https://github.com/cfpb/consumerfinance.gov/blob/cbd6b027516fc1644303f38a42fbeae82e7777bb/cfgov/login/forms.py#L81-L85) be recorded in this case, meaning that, once the lockout period has passed, the user will be able to login.

In this new code, the `super().clean()` will do the same thing, except this new `except` clause will catch the login failure, and record a new login failure, even though it wasn't an actual failure. This can then further invoke the `fa.too_many_attempts()` logic which can reset the lockout period.

So, the problematic sequence is: user gets locked out, gets message to try again "later" (technically, they won't see this message because of the template issue I mentioned, but let's ignore that for now), tries a little later with the correct password, gets the same message - but now they have reset the lockout.

Reply:
Ooooh, that's a good catch! I imagine the same thing would happen with password expiration. I'll write some tests for these scenarios and then fix it.

Reply:
I believe this issue should be fixed in [aeec4d0](https://github.com/cfpb/consumerfinance.gov/pull/7489/commits/aeec4d022ca96dee96efe4e35fbb89aa266e322c).